### ISSUE TITLE
Failure to run '-s ExecutedMutantSelector' on xwiki-commons-xml

### ISSUE THREAD

**Issue by vmassol** [STAMP-project/dspot#292]
In xwiki-commons-xml:

`java -cp /Users/vmassol/dev/dspot/dspot/target/dspot-1.0.5-SNAPSHOT-jar-with-dependencies.jar  fr.inria.stamp.Main --path-to-properties dspot.properties --verbose -i 1 -s ExecutedMutantSelector`

results in:

```
...
2017-12-21 11:07:31,352 DEBUG spoon.Launcher - compiled in 125 ms
2017-12-21 11:07:31,426 INFO fr.inria.diversify.dspot.Amplification - amplification of org.xwiki.xml.html.HTMLUtilsTest (2 test)
2017-12-21 11:07:31,429 DEBUG spoon.Launcher - printing org.xwiki.xml.html.HTMLUtilsTest to target/dspot/tmp_test_sources
2017-12-21 11:07:31,744 INFO fr.inria.diversify.dspot.Amplification - update test testCriterion
2017-12-21 11:07:31,744 INFO fr.inria.diversify.dspot.Amplification - Try to add assertions before amplification
2017-12-21 11:07:31,747 INFO fr.inria.diversify.dspot.assertGenerator.MethodsAssertGenerator - Run tests. (2)
2017-12-21 11:07:31,748 DEBUG spoon.Launcher - printing org.xwiki.xml.html.HTMLUtilsTest to target/dspot/tmp_test_sources
2017-12-21 11:07:31,980 INFO fr.inria.diversify.dspot.assertGenerator.MethodsAssertGenerator - 2 test pass, generating assertion...
2017-12-21 11:07:31,980 INFO fr.inria.diversify.dspot.assertGenerator.MethodsAssertGenerator - Add observations points in passing tests.
2017-12-21 11:07:31,980 INFO fr.inria.diversify.dspot.assertGenerator.MethodsAssertGenerator - Instrumentation...
100% |================================================== ]-
2017-12-21 11:07:31,983 INFO fr.inria.diversify.dspot.assertGenerator.MethodsAssertGenerator - Run instrumented tests. (6)
2017-12-21 11:07:31,984 DEBUG spoon.Launcher - printing org.xwiki.xml.html.HTMLUtilsTest to target/dspot/tmp_test_sources
2017-12-21 11:07:32,207 INFO fr.inria.diversify.dspot.assertGenerator.MethodsAssertGenerator - Generating assertions...
2017-12-21 11:07:32,209 INFO fr.inria.diversify.dspot.assertGenerator.AssertGenerator - 2 new tests with assertions generated
2017-12-21 11:07:32,210 DEBUG spoon.Launcher - printing org.xwiki.xml.html.HTMLUtilsTest to target/dspot/tmp_test_sources
2017-12-21 11:07:32,455 INFO fr.inria.diversify.dspot.Amplification - update test testCriterion
2017-12-21 11:07:32,460 DEBUG spoon.Launcher - printing org.xwiki.xml.XMLUtilsTest to target/dspot/tmp_test_sources
2017-12-21 11:07:32,474 DEBUG spoon.Launcher - building sources: [/Users/vmassol/dev/xwiki/xwiki-commons/xwiki-commons-core/xwiki-commons-xml/target/dspot/tmp_test_sources/org/xwiki/xml/XMLUtilsTest.java, /Users/vmassol/dev/xwiki/xwiki-commons/xwiki-commons-core/xwiki-commons-xml/target/dspot/tmp_test_sources/org/xwiki/xml/html/HTMLUtilsTest.java]
2017-12-21 11:07:32,474 DEBUG spoon.Launcher - build args: [-encoding, UTF-8, -cp, /Users/vmassol/.m2/repository/org/jdom/jdom/1.1.3/jdom-1.1.3.jar:/Users/vmassol/.m2/repository/org/xwiki/commons/xwiki-commons-component-api/9.12-SNAPSHOT/xwiki-commons-component-api-9.12-SNAPSHOT.jar:/Users/vmassol/.m2/repository/org/xwiki/commons/xwiki-commons-stability/9.12-SNAPSHOT/xwiki-commons-stability-9.12-SNAPSHOT.jar:/Users/vmassol/.m2/repository/org/xwiki/commons/xwiki-commons-text/9.12-SNAPSHOT/xwiki-commons-text-9.12-SNAPSHOT.jar:/Users/vmassol/.m2/repository/org/slf4j/slf4j-api/1.7.25/slf4j-api-1.7.25.jar:/Users/vmassol/.m2/repository/javax/inject/javax.inject/1/javax.inject-1.jar:/Users/vmassol/.m2/repository/org/xwiki/commons/xwiki-commons-context/9.12-SNAPSHOT/xwiki-commons-context-9.12-SNAPSHOT.jar:/Users/vmassol/.m2/repository/org/apache/commons/commons-lang3/3.7/commons-lang3-3.7.jar:/Users/vmassol/.m2/repository/net/sourceforge/htmlcleaner/htmlcleaner/2.16/htmlcleaner-2.16.jar:/Users/vmassol/.m2/repository/org/jdom/jdom2/2.0.6/jdom2-2.0.6.jar:/Users/vmassol/.m2/repository/xerces/xercesImpl/2.11.0/xercesImpl-2.11.0.jar:/Users/vmassol/.m2/repository/xml-apis/xml-apis/1.4.01/xml-apis-1.4.01.jar:/Users/vmassol/.m2/repository/net/java/dev/stax-utils/stax-utils/20070216/stax-utils-20070216.jar:/Users/vmassol/.m2/repository/com/fasterxml/woodstox/woodstox-core/5.0.3/woodstox-core-5.0.3.jar:/Users/vmassol/.m2/repository/org/codehaus/woodstox/stax2-api/3.1.4/stax2-api-3.1.4.jar:/Users/vmassol/.m2/repository/org/xwiki/commons/xwiki-commons-tool-test-component/9.12-SNAPSHOT/xwiki-commons-tool-test-component-9.12-SNAPSHOT.jar:/Users/vmassol/.m2/repository/org/xwiki/commons/xwiki-commons-tool-test-simple/9.12-SNAPSHOT/xwiki-commons-tool-test-simple-9.12-SNAPSHOT.jar:/Users/vmassol/.m2/repository/junit/junit/4.12/junit-4.12.jar:/Users/vmassol/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar:/Users/vmassol/.m2/repository/org/junit/jupiter/junit-jupiter-api/5.0.2/junit-jupiter-api-5.0.2.jar:/Users/vmassol/.m2/repository/org/opentest4j/opentest4j/1.0.0/opentest4j-1.0.0.jar:/Users/vmassol/.m2/repository/org/junit/platform/junit-platform-commons/1.0.2/junit-platform-commons-1.0.2.jar:/Users/vmassol/.m2/repository/org/junit/jupiter/junit-jupiter-engine/5.0.2/junit-jupiter-engine-5.0.2.jar:/Users/vmassol/.m2/repository/org/junit/platform/junit-platform-engine/1.0.2/junit-platform-engine-1.0.2.jar:/Users/vmassol/.m2/repository/org/junit/platform/junit-platform-launcher/1.0.2/junit-platform-launcher-1.0.2.jar:/Users/vmassol/.m2/repository/org/junit/vintage/junit-vintage-engine/4.12.2/junit-vintage-engine-4.12.2.jar:/Users/vmassol/.m2/repository/org/hamcrest/hamcrest-library/1.3/hamcrest-library-1.3.jar:/Users/vmassol/.m2/repository/org/jmock/jmock/2.6.0/jmock-2.6.0.jar:/Users/vmassol/.m2/repository/org/jmock/jmock-junit4/2.6.0/jmock-junit4-2.6.0.jar:/Users/vmassol/.m2/repository/commons-io/commons-io/2.6/commons-io-2.6.jar:/Users/vmassol/.m2/repository/org/mockito/mockito-core/2.13.0/mockito-core-2.13.0.jar:/Users/vmassol/.m2/repository/net/bytebuddy/byte-buddy/1.7.9/byte-buddy-1.7.9.jar:/Users/vmassol/.m2/repository/net/bytebuddy/byte-buddy-agent/1.7.9/byte-buddy-agent-1.7.9.jar:/Users/vmassol/.m2/repository/org/objenesis/objenesis/2.6/objenesis-2.6.jar:/Users/vmassol/.m2/repository/ch/qos/logback/logback-classic/1.2.3/logback-classic-1.2.3.jar:/Users/vmassol/.m2/repository/ch/qos/logback/logback-core/1.2.3/logback-core-1.2.3.jar:/Users/vmassol/.m2/repository/org/slf4j/jcl-over-slf4j/1.7.25/jcl-over-slf4j-1.7.25.jar:/Users/vmassol/.m2/repository/org/slf4j/log4j-over-slf4j/1.7.25/log4j-over-slf4j-1.7.25.jar:/Users/vmassol/.m2/repository/org/xwiki/commons/xwiki-commons-observation-api/9.12-SNAPSHOT/xwiki-commons-observation-api-9.12-SNAPSHOT.jar:/Users/vmassol/.m2/repository/jmock/jmock/1.2.0/jmock-1.2.0.jar:/Users/vmassol/.m2/repository/jmock/jmock-cglib/1.2.0/jmock-cglib-1.2.0.jar:/Users/vmassol/.m2/repository/cglib/cglib-nodep/2.1_3/cglib-nodep-2.1_3.jar:/Users/vmassol/.m2/repository/org/xwiki/commons/xwiki-commons-component-default/9.12-SNAPSHOT/xwiki-commons-component-default-9.12-SNAPSHOT.jar:/Users/vmassol/.m2/repository/org/xwiki/commons/xwiki-commons-component-observation/9.12-SNAPSHOT/xwiki-commons-component-observation-9.12-SNAPSHOT.jar:/Users/vmassol/.m2/repository/org/xwiki/commons/xwiki-commons-configuration-api/9.12-SNAPSHOT/xwiki-commons-configuration-api-9.12-SNAPSHOT.jar:/Users/vmassol/.m2/repository/commons-beanutils/commons-beanutils/1.9.3/commons-beanutils-1.9.3.jar:/Users/vmassol/.m2/repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar:../../xwiki-commons-core/xwiki-commons-xml//target/classes:target/dspot/dependencies/:../../xwiki-commons-core/xwiki-commons-xml//target/test-classes:, -1.8, -preserveAllLocals, -noExit, -enableJavadoc, /Users/vmassol/dev/xwiki/xwiki-commons/xwiki-commons-core/xwiki-commons-xml/target/dspot/tmp_test_sources/org/xwiki/xml/XMLUtilsTest.java, /Users/vmassol/dev/xwiki/xwiki-commons/xwiki-commons-core/xwiki-commons-xml/target/dspot/tmp_test_sources/org/xwiki/xml/html/HTMLUtilsTest.java]
Exception in thread "main" java.lang.RuntimeException: spoon.compiler.ModelBuildingException: cleaner cannot be resolved or is not a field at /Users/vmassol/dev/xwiki/xwiki-commons/xwiki-commons-core/xwiki-commons-xml/target/dspot/tmp_test_sources/org/xwiki/xml/XMLUtilsTest.java:41
	at fr.inria.stamp.Main.amplifyAll(Main.java:117)
	at fr.inria.stamp.Main.run(Main.java:61)
	at fr.inria.stamp.Main.run(Main.java:75)
	at fr.inria.stamp.Main.main(Main.java:37)
Caused by: spoon.compiler.ModelBuildingException: cleaner cannot be resolved or is not a field at /Users/vmassol/dev/xwiki/xwiki-commons/xwiki-commons-core/xwiki-commons-xml/target/dspot/tmp_test_sources/org/xwiki/xml/XMLUtilsTest.java:41
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.report(JDTBasedSpoonCompiler.java:581)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.reportProblems(JDTBasedSpoonCompiler.java:562)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.build(JDTBasedSpoonCompiler.java:119)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.build(JDTBasedSpoonCompiler.java:100)
	at spoon.Launcher.buildModel(Launcher.java:709)
	at fr.inria.diversify.dspot.support.DSpotCompiler.compile(DSpotCompiler.java:127)
	at fr.inria.diversify.dspot.selector.ExecutedMutantSelector.selectToKeep(ExecutedMutantSelector.java:100)
	at fr.inria.diversify.dspot.Amplification.preAmplification(Amplification.java:191)
	at fr.inria.diversify.dspot.Amplification.amplification(Amplification.java:75)
	at fr.inria.diversify.dspot.DSpot.amplifyTest(DSpot.java:187)
	at fr.inria.diversify.dspot.DSpot.amplifyTest(DSpot.java:169)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at fr.inria.diversify.dspot.DSpot.amplifyAllTests(DSpot.java:129)
	at fr.inria.stamp.Main.amplifyAll(Main.java:115)
	... 3 more
```

**Comment by danglotb** [comment]
Hi @vmassol 

I made it to quickly i think.

The `ExecutedMutantSelector` was keeping the same reference on the test class under amplification. Also, it was not producing any reports.

See #293, it should fix it.

**Comment by monperrus** [comment]
closed per #292
